Add 'GroupExternalAuthNCheck' directive to prevent mod-authz-external from checking whether Authentication has successfully occurred. + Minor Improvements

bimimicah opened the pull request:
### This is useful for using mod_authz_external as a generic external authorization mechanism, without needing to use Apache's Authentication module (or triggering mod_authn_basic browser login boxes).
- 'GroupExternalAuthNCheck' is set to 'On' by default for compatibility with all existing configurations.
- When set to 'Off', externalgroup_check_authorization() and externalfilegroup_check_authorization() do not perform the Authentication check.
  - Also, if 'user' is not set, we now set it to the empty string to prevent a segfault in the Apache process (from trying to print it to stderr).
- 't' and 'w' are never used in externalfilegroup_check_authorization(), so removed.
- externalgroup_check_authorization() now logs the last failing result code along with the 'User not in Required group.' message.
  - 'code' is initialized to 0 to prevent a compiler warning.

bimimicah commented:
Added an issue for this pull request: #29

phokz commented:
Hi, your changes seems to be fine. I am about to merge them.
But I think there should be mention in docs as well, maybe a example apache2 config file.

Best regards,
J.

bimimicah commented:
I added a small new section to the INSTALL file, next to the other authz/group-related section.

Would you be interested in re-opening the wiki section for this repository?
I took the wiki branch from the Google Code import and made it the wiki repo on my fork; I fixed some links, added a sidebar file, and some other things.
I personally think the wiki is much more usable than the text files..
